MinecraftClient/Protocol/MicrosoftAuthentication.cs

```diff
@@ -4,9 +4,104 @@
 using System.Text;
 using System.Text.RegularExpressions;
 using System.Collections.Specialized;
+using System.Diagnostics;
 
 namespace MinecraftClient.Protocol
 {
+    static class Microsoft
+    {
+        private static readonly string clientId = "54473e32-df8f-42e9-a649-9419b0dab9d3";
+        /// <summary>
+        /// Client secret has limited lifetime. Current will be expired on 5/6/2022
+        /// </summary>
+        private static readonly string clientSecret = Uri.EscapeDataString("MbH7Q~~UPIybhpAELRKMjSXO6Ar_9A5w-uUw5");
```

Comment:
Can we generate one for each user and store it locally?

Reply:
No, client secret is obtained from Azure application setting. I think it is not possible to obtain more secret programmatically. But never mind, client secret is no longer needed after I changed the Azure application type from web to desktop app.

Reply:
If client secret cannot be changed, it's indeed pointless to have it inside openly available source code. Good thing you can change to a mode that doesn't have it 😄